validator/sawtooth_validator/journal/completer.py

```diff
@@ -75,6 +88,22 @@ def add_batch(self, batch):
         self.batch_cache[batch.header_signature] = batch
         self._on_batch_received(batch)
 
+    def get_block(self, block_id):
+        if block_id in self.block_cache:
+            return self.block_cache[block_id]
+        return None
+
+    def get_batch(self, batch_id=None, transaction_id=None):
```

Comment:
Consider splitting these into two methods get_batch and get_batch_by_transaction